lib/src/rules/sort_public_methods_before_private_methods.dart

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2022, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+
+import '../analyzer.dart';
+
+const _desc = r'Define public methods before private methods.';
+
+const _details = r'''
+**DO** define public methods before private methods.
+
+This makes the code uniform across multiple classes
+and it makes it faster to find specific methods in a class. 
+
+**BAD:**
+```dart
+class A {
+  int a() => 0;
+  int _b() => 0;
+  int c() => 0;
+}
+```
+
+**GOOD:**
+```dart
+class A {
+  int a() => 0;
+  int c() => 0;
+  int _b() => 0;
+}
+```
+
+''';
+
+class SortPublicMethodsBeforePrivateMethods extends LintRule {
+  SortPublicMethodsBeforePrivateMethods()
+      : super(
+            name: 'sort_public_methods_before_private_methods',
```

Comment:
Could you please elaborate why this also applies for other members?
As I understand it in line 66 `if (member is MethodDeclaration) {` this only applies for method declarations and not for other declarations, right?.
But I may have also missed something, because this is my first attempt to writing a lint rule.

Reply:
For example this piece gets not complained about.
```
class A {
  final _y = 0;
  final x = 0;
}
```

Reply:
Regarding the naming I like the idea to shorten the name.
If this only applies to methods I would suggest `sort_private_methods_last`, when this applies to all members your suggestion looks good.

I will change the name when we decided if this only applies to methods or all members.

Reply:
You're right, that comment was outdated. I hadn't noticed the test for method declarations on the first read through, so when I went back to start commenting I had the wrong understanding, and forgot to update the comment when I did see the test. Sorry for the noise.

Reply:
I renamed the rule to `sort_private_methods_last`.